fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java

```diff
@@ -725,9 +698,20 @@ public static Type transformTypeForMv(Type srcType) {
                 newType = ScalarType.createType(srcType.getPrimitiveType());
             }
         } else if (srcType.isArrayType()) {
-            newType = new ArrayType(transformTypeForMv(((ArrayType) srcType).getItemType()));
+            newType = new ArrayType(transformTableColumnType(((ArrayType) srcType).getItemType(), convertDouble));
+        } else if (srcType.isMapType()) {
+            newType = new MapType(transformTableColumnType(((MapType) srcType).getKeyType(), convertDouble),
+                    transformTableColumnType(((MapType) srcType).getValueType(), convertDouble));
+        } else if (srcType.isStructType()) {
+            StructType structType = (StructType) srcType;
+            List<StructField> mappingFields = structType.getFields().stream()
+                    .map(x -> new StructField(x.getName(), transformTableColumnType(x.getType(), convertDouble),
+                            x.getComment()))
+                    .collect(Collectors.toList());
+            newType = new StructType(mappingFields, structType.isNamed());
         } else {
-            throw new SemanticException("Unsupported Mv transform type: %s", srcType);
+            // Default behavior is keep source type
+            return newType;
```

Comment:
So MV/CTAS will support all types: like hll/bitmap/percentile? If so, may add tests for those?

 we can remove the restrictions for MV/CTAS in the doc later.